plugins/core/gs-appstream.c

```diff
@@ -1408,6 +1402,25 @@ gs_appstream_component_add_keyword (XbBuilderNode *component, const gchar *str)
 	}
 }
 
+void
+gs_appstream_component_add_kudo (XbBuilderNode *component, const gchar *str)
+{
+	g_autoptr(XbBuilderNode) kudos = NULL;
+	g_autoptr(XbBuilderNode) kudo = NULL;
+
+	/* create <kudos> if it does not already exist */
+	kudos = xb_builder_node_get_child (component, "kudos", NULL);
+	if (kudos == NULL)
+		kudos = xb_builder_node_insert (component, "kudos", NULL);
+
+	/* create <kudo>str<kudo> if it does not already exist */
+	kudo = xb_builder_node_get_child (kudo, "kudo", str);
```

Comment:
This was causing the warnings. Fixed up locally. 
```suggestion
	kudo = xb_builder_node_get_child (kudos, "kudo", str);
```